src/com/massivecraft/factions/engine/EnginePermBuild.java

```diff
@@ -209,6 +213,27 @@ public void buildEntity(EntityDamageByEntityEvent event)
 		build(player, entity.getLocation().getBlock(), event);
 	}
 	
+	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+	public void combustEntity(EntityCombustByEntityEvent event) {
+		
+		// If a burning projectile ...
+		if (!(event.getCombuster() instanceof Projectile)) return;
+		Projectile entityProjectile = (Projectile)event.getCombuster();
+		
+		// ... fired by a player ...
+		ProjectileSource projectileSource = entityProjectile.getShooter();
+		if (MUtil.isntPlayer(projectileSource)) return;
+		
+		// ... and hits an entity which is edited on damage (and thus likely to burn) ...
+		Entity entityTarget = event.getEntity();
+		if (!EnumerationUtil.isEntityTypeEditOnDamage(entityTarget.getType())) return;
```

Comment:
Does this not need a null check for `entityTarget`?

Reply:
I'd coded under the assumption that, given the nature of the event, getEntity would never be null (unlike the earlier commit, where the Event could fire for either a Block or an Entity).

If I was mistaken, I certainly wouldn't have expected such an edge case.